src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -697,11 +695,58 @@ public bool IsReadOnly()
 
         public void Assign(string expression)
         {
+            this.VerifyNotDisposed();
+
             _engine.DebuggedProcess.WorkerThread.RunOperation(async () =>
             {
                 _engine.DebuggedProcess.FlushBreakStateData();
                 Value = await _engine.DebuggedProcess.MICommandFactory.VarAssign(_internalName, expression);
             });
         }
+
+        #region IDisposable Implementation
+
+        private bool _isDisposed = false;
+
+        private void VerifyNotDisposed()
+        {
+            if (this._isDisposed)
+            {
+                throw new ObjectDisposedException(this.GetType().FullName);
+            }
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(true);
+
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool isDisposing)
+        {
+            this._isDisposed = true;
```

Comment:
Potentially, but due to the use of async/await, I think this is the best we can reasonably do.